include/boost/spirit/home/classic/iterator/multi_pass.hpp

```diff
@@ -770,14 +769,13 @@ struct iterator_base_creator
 {
     typedef typename InputPolicyT::BOOST_NESTED_TEMPLATE inner<InputT> input_t;
 
-    typedef boost::iterator
-    <
-        std::forward_iterator_tag,
-        typename input_t::value_type,
-        typename input_t::difference_type,
-        typename input_t::pointer,
-        typename input_t::reference
-    > type;
+    struct type {
```

Comment:
It looks like the comment above is obsolete because of https://github.com/boostorg/core/commit/bca30b2ec7f41e6ece057b46d1358f0506c904d3.

Reply:
No, it's not. Because of the change by that commit, inheriting from boost::iterator or referring to its members is just the same as inheriting from or using std::iterator. This is deemed deprecated now since C++17 and the latest version of MSVC spits out nasty warnings because of that. On top of that, Boost's version of iterator.hpp is deprecated as well.

Reply:
```
 // Meta-function to generate a std::iterator<> base class for multi_pass. This
 //  is used mainly to improve conformance of compilers not supporting PTS
 //  and thus relying on inheritance to recognize an iterator.
```
I do not think this part makes any sense after the change.

```
 // We are using boost::iterator<> because it offers an automatic workaround
 //  for broken std::iterator<> implementations.
```
And this since https://github.com/boostorg/core/commit/bca30b2ec7f41e6ece057b46d1358f0506c904d3.

Reply:
Ah, *that's* your objection. Nevermind - I'll change that asap.